### ISSUE TITLE
TypeError: int() argument must be a string or a number, not 'changectx'

### ISSUE THREAD

**Issue by eigengrau** [felipec/git-remote-hg#72]
With git-remote-hg (HEAD) and mercurial 4.6, git fetch yields:

```
  File "/usr/bin/git-remote-hg", line 1359, in <module>
    sys.exit(main(sys.argv))
  File "/usr/bin/git-remote-hg", line 1343, in main
    do_import(parser)
  File "/usr/bin/git-remote-hg", line 742, in do_import
    export_branch(repo, branch)
  File "/usr/bin/git-remote-hg", line 634, in export_branch
    export_ref(repo, branch, 'branches', head)
  File "/usr/bin/git-remote-hg", line 502, in export_ref
    revs = gitrange(repo, tip, head)
  File "/usr/bin/git-remote-hg", line 469, in gitrange
    pending = set([int(b)])
TypeError: int() argument must be a string or a number, not 'changectx'
```

I haven’t dug deeper, but I suspect mercurial is now encapsulating revisions in `changectx` objects, where it passed hashes before.

In my case, it looks like unwrapping the `a`, `b` arguments to `gitrange` resolves this.

```
def gitrange(repo, a, b):
   a = a.rev()
   b = b.rev()
   …
```

I’d open a PR, but I suspect you’ll want to guard this for backwards compatibility somehow.

**Comment by matthewfl** [comment]
I just tested this and it works as a fix for me.

I just added:
```python
if hasattr(a, 'rev'):
    a = a.rev()
    b = b.rev()
```
To the `getrange` function to check if there is a rev attribute/function that it can call.  I am not 100% sure what was getting passed before or if that is sufficient to guard calling the `rev` method.

**Comment by dliessi** [comment]
This solves the problem also for me.
I would check the presence of `rev` in both `a` and `b`.

**Comment by karianna** [comment]
@matthewfl patch worked for me also

**Comment by mnauw** [comment]
These issues are due to changes in recent Mercurial's internal API.  They have also been reported as issues in this [fork](https://github.com/mnauw/git-remote-hg) and are already fixed there (along with many other ones), see also generally #69.

**Comment by devinrhode2** [comment]
fork worked for me @mnauw, thank you guys very much!!!!

**Comment by felipec** [comment]
> ```
> def gitrange(repo, a, b):
>    a = a.rev()
>    b = b.rev()
>    …
> ```

This is the correct fix: 00e95fd8df745db297eab035b31af8a518627bb1.

Should be fixed in v0.4.

**Comment by felipec** [comment]
> Still doesn't work for me:
>
> Even tried forks of this repo, all are broken.

Your's a separate issue fixed in c95fba3c184128599f73c45ca6a3c32e0bd7f87e.

Both of those issues should work in v0.4.
